lib/Raven/Serializer.php

```diff
@@ -109,6 +109,12 @@ protected function serializeValue($value)
         if (is_null($value) || is_bool($value) || is_float($value) || is_integer($value)) {
             return $value;
         } elseif (is_object($value) || gettype($value) == 'object') {
+            // If the value is a callable and expects no parameters, we evaluate the
+            // callable value by calling it.
+            if (is_callable($value) && Raven_Util::getCallableParamNum($value) === 0) {
+                return $value();
```

Comment:
In this context I would say a compromised system would not be relying on the Sentry client to be available. A developer who understands the basic issue with user submitted data/content, server security, upload authz, and common sense will not be opening up holes with this.

It would be on the hands of the Sentry integrator to create this security hole, by allowing a non-predefined callable to be passed to the serializer. The way this Sentry client is integrated most of the time means there is an initial setup during which the context is laid out.

I can understand the risk being applicable in situations like this

```php
<?php

$just_call_me_once = function () {
    $methodname = $_REQUEST['method_to_use'];

    // method_to_use === 'die' (or similar)

    return $methodname();
};

$client->extra_context([
    'my_value' => $just_call_me_once
]);
```

Which shows the developer doing something a developer should never be doing: unsanitized input from user.

If someone can inject a callable (such as `$just_call_me_once`) _and_ tie it to the Sentry client context (via `$client->extra_context()` for instance), I guess the work to achieve that could be severed to a fraction by just calling the malicious code instead of tying it to Sentry and then triggering an error that triggers serialization in Sentry.

Of course I could be way off on this, I'm not a security specialist. But if I wanted to cause harm via my own code I would not jump through rings and hoops by attemping to get Sentry to serialize something.

The main issue would (in my opinion) be developers who do not understand the harm what dynamically selected values/functions/classes based on some request specific parameters can cause. If this is too big of a risk in the eyes of package maintainers I do understand. All suggestions welcome on how this can be implemented in the safest way possible (maybe a flag `allow_callable_serialization` or something for those who understand the risks involved).

Please let me know if there is something I'm clearly missing about this and if I'm making my case for nothing. Thank you for opening the discussion @nokitakaze. :)